app/workers/start_of_cycle_notification_worker.rb

```diff
@@ -0,0 +1,49 @@
+class StartOfCycleNotificationWorker
+  include Sidekiq::Worker
+
+  def perform(service, batch_size = 500)
```

Comment:
maybe we could calculate the batch number from the provider final count on the day, so we could divvy those up equally between the hours?